packages/style-engine/class-wp-style-engine.php

```diff
@@ -345,6 +351,21 @@ protected static function get_css( $style_value, $style_definition, $should_retu
 		// for styles such as margins and padding.
 		if ( is_array( $style_value ) ) {
 			foreach ( $style_value as $key => $value ) {
+				if ( is_string( $value ) && strpos( $value, 'var:' ) !== false ) {
+					if ( $should_return_css_vars && ! empty( $style_definition['css_vars'] ) ) {
+						foreach ( $style_definition['css_vars'] as $property_key => $css_var_pattern ) {
+							$slug = static::get_slug_from_preset_value( $value, $property_key );
```

Comment:
Is there a way to abstract this given that it's almost the same as the css var parsing above in the same method?

If it's too messy, an alternative might be to provide a custom `value_func`. See `get_css_individual_property_rules()` below.

Reply:
I was just typing nearly the exactly the same comment 😆!

Other than the duplication, this approach looks good to me @glendaviesnz, I reckon a small method for parsing the value to the `var()` rule would do the trick 👍

Reply:
good point, will try and DRY that up